### PR TITLE
feat(@types/emoji-js) create types

### PR DESCRIPTION
**Pull request by jasondibenedetto**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/emoji-js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// Type definitions for emoji-js 3.4
+// Project: https://github.com/iamcal/js-emoji
+// Definitions by: Jason Di Benedetto <https://github.com/me>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+export default class EmojiConvertor {
```
Comment (armanio123) [inline types/emoji-js/index.d.ts]: I don't think the library exports a `default`. Please remove this.

File: types/emoji-js/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    constructor();
```
Comment (armanio123) [inline types/emoji-js/index.d.ts]: There's no point on defining an empty constructor. You can remove this line.

File: types/emoji-js/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+// Type definitions for emoji-js 3.4
+// Project: https://github.com/iamcal/js-emoji
+// Definitions by: Jason Di Benedetto <https://github.com/me>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
```
Comment (armanio123) [inline types/emoji-js/index.d.ts]: Library appears to be an UMD module. You also need to define `export as namespace EmojiConvertor;`

File: types/emoji-js/emoji-js-tests.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import EmojiConvertor from 'emoji-js';
```
Comment (armanio123) [inline types/emoji-js/emoji-js-tests.ts]: Since there's no `default`, you need to change this import to `import EmojiConvertor = require('emoji-js');`

**Review by armanio123 — CHANGES_REQUESTED** [review]

**Comment by jasondibenedetto** [comment]
Thanks for the feedback @armanio123. I have amended my commit with your suggestions.

**Review by armanio123 — APPROVED** [review]

**Pull request merged**